src/accountTransactions.ts

```diff
@@ -1,45 +1,58 @@
 import { Buffer } from 'buffer/';
-import { encodeWord64 } from './serializationHelpers';
+import { encodeWord64, encodeMemo } from './serializationHelpers';
 import {
     AccountTransactionPayload,
     AccountTransactionType,
-    SimpleTransfer,
+    SimpleTransferPayload,
+    SimpleTransferWithMemoPayload
 } from './types';
 
-interface AccountTransactionHandler {
-    serialize: (payload: AccountTransactionPayload) => Buffer;
-    getBaseEnergyCost: (payload?: AccountTransactionPayload) => bigint;
+interface AccountTransactionHandler<PayloadType extends AccountTransactionPayload = AccountTransactionPayload> {
+    serialize: (payload: PayloadType) => Buffer;
+    getBaseEnergyCost: (payload?: PayloadType) => bigint;
 }
 
-export class SimpleTransferHandler implements AccountTransactionHandler {
+export class SimpleTransferHandler implements AccountTransactionHandler<SimpleTransferWithMemoPayload> {
     getBaseEnergyCost(): bigint {
         return 300n;
     }
 
-    serialize(transfer: SimpleTransfer): Buffer {
+    serialize(transfer: SimpleTransferPayload): Buffer {
         const serializedToAddress = transfer.toAddress.decodedAddress;
         const serializedAmount = encodeWord64(transfer.amount.microGtuAmount);
         return Buffer.concat([serializedToAddress, serializedAmount]);
     }
 }
 
-export function getAccountTransactionHandler(
-    type: AccountTransactionType
-): AccountTransactionHandler {
-    const accountTransactionHandlerMap = new Map<
-        AccountTransactionType,
-        AccountTransactionHandler
-    >();
-    accountTransactionHandlerMap.set(
-        AccountTransactionType.SimpleTransfer,
-        new SimpleTransferHandler()
-    );
+export class SimpleTransferWithMemoHandler implements AccountTransactionHandler<SimpleTransferWithMemoPayload> {
+    getBaseEnergyCost(): bigint {
+        return 300n;
+    }
 
-    const handler = accountTransactionHandlerMap.get(type);
-    if (!handler) {
-        throw new Error(
-            'The handler map is missing the provided type: ' + type
+    serialize(transfer: SimpleTransferWithMemoPayload): Buffer {
+        const serializedToAddress = transfer.toAddress.decodedAddress;
+        const serializedAmount = encodeWord64(transfer.amount.microGtuAmount);
+        const serializedMemo = encodeMemo(
+            (transfer as SimpleTransferWithMemoPayload).memo
```

Comment:
`transfer as SimpleTransferWithMemoPayload` seems redundant when you have
```
serialize(transfer: SimpleTransferWithMemoPayload)
```

Reply:
It is, I was also able to refactor more away.